plugin/checkout_handlers_test.go

```diff
@@ -101,3 +114,75 @@ func TestValidateInputs(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestPasswordHandlerInterfaceFulfillment(t *testing.T) {
+	ctx, storage, serviceAccountName, checkOut := setup()
+
+	passwordHandler := &PasswordHandler{
+		client: &fakeSecretsClient{},
+		child:  &fakeCheckOutHandler{},
+	}
+
+	// There should be no error during check-out.
+	if err := passwordHandler.CheckOut(ctx, storage, serviceAccountName, checkOut); err != nil {
+		t.Fatal(err)
+	}
+
+	// The password should get rotated successfully during check-in.
+	origPassword, err := retrievePassword(ctx, storage, serviceAccountName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if origPassword != "" {
+		t.Fatal("expected empty password")
+	}
+	if err := passwordHandler.CheckIn(ctx, storage, serviceAccountName); err != nil {
+		t.Fatal(err)
+	}
+	currPassword, err := retrievePassword(ctx, storage, serviceAccountName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if currPassword == origPassword {
+		t.Fatal("expected new password, but received none")
+	}
+
+	// There should be no error during delete and the password should be deleted.
+	if err := passwordHandler.Delete(ctx, storage, serviceAccountName); err != nil {
+		t.Fatal(err)
+	}
+
+	// There should be no error during status.
+	currPassword, err = retrievePassword(ctx, storage, serviceAccountName)
+	if err != nil {
```

Comment:
I'm surprised this isn't an error. Given that delete "cleans up anything we were tracking from the service account", I'd have thought there would be a not found error at this point in the test.

Reply:
That's fine. We kind of go back and forth in the code. Storage returns nil, nil if there's no entry, so I was following that pattern here. I actually prefer the pattern of returning a strongly typed error, so I'm happy to switch over to it.

Reply:
Pushed an update with this.